File: sample-conference-videochat/sample-conference-videochat/Components/Settings/VideoSettingsViewController/BaseSettingsCell.m
//
//  BaseSettingsCell.m
//  sample-conference-videochat
//
//  Created by Injoit on 30/09/15.
//  Copyright © 2015 QuickBlox Team. All rights reserved.
//

#import "BaseSettingsCell.h"

@implementation BaseSettingsCell

+ (NSString *)identifier {
    return NSStringFromClass([self class]);
}

+ (UINib *)nib {
    return [UINib nibWithNibName:[self identifier] bundle:nil];
}

- (void)setModel:(BaseItemModel *)model {
    
    _model = model;
    self.label.text = model.title;
}

@end

File: sample-conference-videochat-swift/sample-conference-videochat-swift/sample-conference-videochat-swift/Components/Settings/BaseSettingsViewController/BaseSettingsCell.swift
//
//  BaseSettingsCell.swift
//  sample-conference-videochat-swift
//
//  Created by Vladimir Nybozhinsky on 22.10.2018.
//  Copyright © 2018 QuickBlox. All rights reserved.
//

import UIKit

protocol SettingsCellDelegate: class {
    func cell(_ cell: BaseSettingsCell?, didChageModel model: BaseItemModel?)
}

class BaseSettingsCell: UITableViewCell {
    @IBOutlet weak var label: UILabel!
    var model: BaseItemModel? {
        didSet {
            label.text = model?.title
        }
    }
    weak var delegate: SettingsCellDelegate?
    
    class func identifier() -> String? {
        return NSStringFromClass(BaseSettingsCell.self).components(separatedBy: ".").last!
    }
    
    class func nib() -> UINib? {
        return UINib(nibName: NSStringFromClass(BaseSettingsCell.self), bundle: nil)
    }
}
